myway/src/main/java/com/hmel/myway/dao/blogic/services/BaseHibernateDAO.java

```diff
@@ -51,14 +48,23 @@ public T findOne(P id) throws PhoneDictionaryException {
 
 	}
 
+	/*
+	 * int firstResult, int maxResults
+	 */
 	@SuppressWarnings("unchecked")
 	@Transactional(value = "transactionManager")
-	public List<T> findAll() throws PhoneDictionaryException {
+	public List<T> findAll(int firstResult, int maxResults) throws PhoneDictionaryException, IllegalArgumentException{
 		logger.info("findAll CALLING");
 		getCurrentSession().beginTransaction();
+		if(firstResult<0){
+			throw new IllegalArgumentException("first result must be >=0");
+		}
+		if(maxResults<=0 || maxResults<firstResult){
```

Comment:
conditions
firstResult=0;
maxResults=0;
so 59 line all is ok
if 62 will be only  maxResults<firstResult - all is ok
but if we will use what we have now it will show Exception
if we can put everuthing in method we must check all parameters. so it is sense to check also this. user can put 0 and 0 and you will execute query without sense